### ISSUE TITLE
Segfault when writing Array{Union{T, Missings.Missing}, 1}

### ISSUE THREAD

**Issue by JockLawrie** [JuliaData/CSV.jl#146]
I accidentally tried to write an Array using `CSV.write`.
When `eltype(arr)` is a primitive type such as `Int` I get an error as expected.
When `eltype(arr)` is `Union{Int, Missings.Missing}` I get a segfault, illegal instruction.
Example below.

```julia
using Missings
using CSV

arr = [11]
CSV.write("test.csv", arr)  # Error as expected

arr = missings(Int, 1)
arr[1] = 11
CSV.write("test.csv", arr)  # Segfault
```

**Comment by nalimilan** [comment]
Confirmed on 0.6.2, but luckily it's fixed on 0.7.

**Comment by JockLawrie** [comment]
Cool.
Nothing to do - writing an array was bad code on my part anyway.
Will wait for 0.7.

Thanks again!
Jock
